bin/run-java-shopping-cart-test.sh

```diff
@@ -0,0 +1,250 @@
+spec:
+  type: Postgres
+  deployment: Unmanaged
+  config:
+    service: $POSTGRES_SERVICE
+    credentials:
+      database: $POSTGRES_DATABASE
+      username: $POSTGRES_USERNAME
+      password: $POSTGRES_PASSWORD
+---
+apiVersion: cloudstate.io/v1alpha1
+kind: StatefulService
+metadata:
+  name: $statefulservice
+spec:
+  datastore:
+    name: $statefulstore
+  containers:
+    - image: cloudstatedev/java-shopping-cart:dev
+YAML
+;;
+
+  cassandra ) # deploy the shopping-cart with cassandra store
+
+  statefulstore="cassandra"
+  statefulservice="shopping-cart-$statefulstore"
+
+  [ -f cassandra-env.sh ] && source cassandra-env.sh
+
+  kubectl apply -f - <<YAML
+apiVersion: cloudstate.io/v1alpha1
+kind: StatefulStore
+metadata:
+  name: $statefulstore
+spec:
+  type: Cassandra
+  deployment: Unmanaged
+  config:
+    service: $CASSANDRA_SERVICE
+    credentials:
+      username: $CASSANDRA_USERNAME
+      password: $CASSANDRA_PASSWORD
+---
+apiVersion: cloudstate.io/v1alpha1
+kind: StatefulService
+metadata:
+  name: $statefulservice
+spec:
+  datastore:
+    name: $statefulstore
+    config:
+      keyspace: shoppingcart
+  containers:
+    - image: cloudstatedev/java-shopping-cart:dev
+YAML
+;;
+
+  inmemory | * ) # deploy the shopping-cart with in-memory store
+
+  statefulstore="inmemory"
+  statefulservice="shopping-cart-$statefulstore"
+
+  kubectl apply -f - <<YAML
+apiVersion: cloudstate.io/v1alpha1
+kind: StatefulStore
+metadata:
+  name: $statefulstore
+spec:
+  type: InMemory
+---
+apiVersion: cloudstate.io/v1alpha1
+kind: StatefulService
+metadata:
+  name: $statefulservice
+spec:
+  datastore:
+    name: $statefulstore
+  containers:
+    - image: cloudstatedev/java-shopping-cart:dev
+YAML
+;;
+
+esac
+
+deployment="$statefulservice-deployment"
+
+# Can only kubectl wait when the resource already exists
+echo
+echo "Waiting for deployment to be created..."
+attempts=10
+until kubectl get deployment $deployment &> /dev/null || [ $attempts -eq 0 ] ; do
+  ((attempts--))
+  sleep 1
+done
+kubectl get deployment $deployment
+
+function fail_with_details {
+  echo
+  echo "=== Operator logs ==="
+  echo
+  kubectl logs -l app=cloudstate-operator -n cloudstate --tail=-1
+  echo
+  echo "=== Deployment description ==="
+  echo
+  kubectl describe deployment/$deployment
+  echo
+  echo "=== Pods description ==="
+  echo
+  kubectl describe pods
+  echo
+  echo "=== Proxy logs ==="
+  echo
+  kubectl logs -l app=$statefulservice -c akka-sidecar --tail=-1
+  echo
+  echo "=== User container logs ==="
+  echo
+  kubectl logs -l app=$statefulservice -c user-container --tail=-1
+  exit 1
+}
+
+# Wait for the deployment to be available
+echo
+echo "Waiting for deployment to be ready..."
+kubectl wait --for=condition=available --timeout=1m deployment/$deployment || fail_with_details
+kubectl get deployment $deployment
+
+# Scale up the deployment, to test with akka clustering
+# Travis can fail with 3 nodes, only scale to 2
```

Comment:
Perhaps we should remove the smoke tests from the travis build?

Reply:
Yeah, we could run all of these on CircleCI instead. Looks like we've already extended this month's build credits, so we may need to trim back the testing. We could see how it goes.

Reply:
Updated in #410. Tried the default machine size first, but that can't scale to 3 nodes, like Travis. The large machine (which we pay for) works fine.